### PR TITLE
Add trunk blocks & associated logic

### PR DESCRIPTION
**Pull request by Qwertygiy**
Similar to the existing Door block, Trunks are chests that are two blocks wide (and thus have twice as much storage capacity!)

image

To test, enter a world with the "develop" branch of CoreAssets and [this PR,](https://github.com/Terasology/CoreBlocks/pull/2) or the relevant branch, of CoreBlocks.

Open console with F1, enter `give trunk`, and right-click upon a block surface to place the trunk. Once placed, press E to interact and open the trunk, and access its 60 storage slots.

image

To Do:

- [x] Create proper item icon or rendering. Currently uses the door texture as a placeholder.
- [x] Correct block placement. The right block occasionally winds up being placed in the left spot.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Qwertygiy** [comment]
image

File: modules/Core/src/main/java/org/terasology/core/logic/trunk/TrunkComponent.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+package org.terasology.core.logic.trunk;
+
+import org.terasology.audio.StaticSound;
+import org.terasology.entitySystem.Component;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.math.Side;
+import org.terasology.world.block.family.BlockFamily;
+
+/**
```
Comment (skaldarnar) [inline modules/Core/src/main/java/org/terasology/core/logic/trunk/TrunkComponent.java]: Going ahead with good example - let's add Javadoc 😉 
```suggestion
/**
  * Trunks are chests that are two blocks wide, with respective sounds for opening and closing the trunk.
```

File: modules/Core/src/main/java/org/terasology/core/logic/trunk/TrunkSystem.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    private EntityManager entityManager;
+    @In
+    private AudioManager audioManager;
+    @In
+    private InventoryManager inventoryManager;
+    @In
+    private EntitySystemLibrary entitySystemLibrary;
+
+    @ReceiveEvent(components = {TrunkComponent.class, ItemComponent.class})
+    public void placeTrunk(ActivateEvent event, EntityRef entity) {
```
Comment (skaldarnar) [inline modules/Core/src/main/java/org/terasology/core/logic/trunk/TrunkSystem.java]: Could you please split this up into smaller (private) methods to slim this method down a bit?

Comment (skaldarnar) [reply modules/Core/src/main/java/org/terasology/core/logic/trunk/TrunkSystem.java]: could also be done in a follow-up PR, tbh.

**Pull request merged**